common/flatpak-dir.c

```diff
@@ -3625,8 +3660,15 @@ flatpak_dir_pull (FlatpakDir          *self,
                   subdirs_arg ? (const char **)subdirs_arg->pdata : NULL,
                   ref, rev, results, flatpak_flags, flags,
                   progress,
-                  cancellable, error))
+                  cancellable, &my_error))
     {
+      if (g_error_matches (my_error, G_IO_ERROR, G_FILE_ERROR_NOSPC) ||
+          flatpak_dir_check_for_free_space (repo))
+        {
+          flatpak_dir_cache_clean (repo);
```

Comment:
This feels like something that should live in libostree by default; deleting everything in the repo's `tmp` directory is also going to break anything else concurrently using that directory (although in practice other things would probably also fail with `ENOSPC` too).

Reply:
Living in ostree would probably also make it easier to correctly propagate the ENOSPC error.

Reply:
I wonder here. The logic here is that, the default action is to clean up (in case of failure), unless the user has free-space above a certain threshold. For e.g. keep caches only of free space is above 5GBs.
The primary reason I chose this is to work around the ENOSPC error (and it's problem in error-reporting).

Reply:
Hi, made a related PR to this in ostree. Please comment on it. https://github.com/ostreedev/ostree/pull/1602